src/luv.c

```diff
@@ -508,6 +508,39 @@ static void luv_handle_init(lua_State* L) {
   lua_setfield(L, LUA_REGISTRYINDEX, "uv_stream");
 }
 
+// Call lua function, will pop nargs values from top of vm stack and push some
+// values according to nresults. When error occurs, it will print error message
+// to stderr, and memory allocation error will cause exit.
+static int luv_cfpcall(lua_State* L, int nargs, int nresult) {
+  int ret, errfunc;
+
+  // Get the traceback function in case of error
+  lua_pushcfunction(L, luv_traceback);
+  errfunc = lua_gettop(L);
+  // And insert it before the function and args
+  lua_insert(L, -2 - nargs);
+  errfunc -= (nargs+1);
+
+  ret = lua_pcall(L, nargs, nresult, errfunc);
+  switch (ret) {
+  case LUA_OK:
+    break;
+  case LUA_ERRMEM:
+    fprintf(stderr, "System Error: %s\n", lua_tostring(L, -1));
+    exit(-1);
+    break;
+  case LUA_ERRRUN:
+  case LUA_ERRSYNTAX:
+  case LUA_ERRERR:
+  default:
+    fprintf(stderr, "Uncaught Error: %s\n", lua_tostring(L, -1));
+    lua_pop(L, 1);
```

Comment:
shouldn't we return `0` in case of error, as no results will be pushed on the stack? or maybe `-1` as an indicator.  (doesn't matter now, but in case this is used with return values later, it is good to make the interface proper).